cpp/src/arrow/table-test.cc

```diff
@@ -498,6 +498,17 @@ TEST_F(TestTable, SetColumn) {
   ASSERT_TRUE(result->Equals(*expected));
 }
 
+TEST_F(TestTable, RenameColumns) {
+  MakeExample1(10);
+  auto table = Table::Make(schema_, columns_);
+  auto names = table->ColumnNames();
+  ASSERT_EQ(names, std::vector<std::string>({"f0", "f1", "f2"}));
```

Comment:
Note that gmock allows shorter syntax:

`EXPECT_THAT(table->ColumnNames(), ElementsAre("f0", "f1", "f2"));`

No need for temporary, and you'll get nice debug errors :)